app/context/server_display_name/index.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ComponentType, createContext} from 'react';
+
+type Props = {
+    displayName: string;
+    children: React.ReactNode;
+}
+
+type WithServerDisplayNameProps = {
+    serverDisplayName: string;
+}
+
+const ServerDisplayNameContext = createContext<string>('');
+const {Provider, Consumer} = ServerDisplayNameContext;
+
+function ServerDisplayNameProvider({displayName, children}: Props) {
+    return (
+        <Provider value={displayName}>{children}</Provider>
+    );
+}
+
+export function withServerDisplayName<T extends WithServerDisplayNameProps>(Component: ComponentType<T>): ComponentType<T> {
```

Comment:
I get the reason why you are doing this, but wonder if we can set this value in an object alongside the serverUrl Provider and have the hooks `useServerUrl` and `useServerDisplayName` just return the value from the object, do you see a reason why not doing it the way I'm proposing ?